README.md

```diff
@@ -16,6 +16,328 @@ K8ssandra 1.x is configured, packaged, and deployed via Helm charts. Those Helm
+
+If you are running in a cloud provider, you can get routable IPs by installing the Kubernetes clusters in the same VPC.
+
+If you run multiple kind clusters locally, you will have routable pod IPs assuming that they run on the same Docker network which is normally the case. We leverage this for our multi-cluster e2e tests.
+
+## Architecture
+K8sandra Operator consists of a control plane and a data plane. Simply put the data plane deploys and manages pods. The control plane does not deploy or manage pods. The control plane should be installed in only one cluster, i.e., the control plane cluster. The data plane can be installed on any number of clusters.
+
+**Note:** The control plane cluster can also function as a data plane cluster.
+
+**TODO:** Add architecture diagram
+
+## Connecting to remote clusters
+The control plane needs to establish client connections to remote cluster where the data plane runs. Credentials are provided via a [kubeconfig](https://kubernetes.io/docs/tasks/access-application-cluster/configure-access-multiple-clusters/) file that is stored in a Sscret. That secret is then referenced via a `ClientConfig` custom resource.
+
+### Creating a ClientConfig
+First, we need to create a secret that the kubeconfig file. 
+
+Do not use `$HOME/.kube/config` as it may contain keys to cluster that you do not intend to expose. Instead generate separate kubeconfig files for the clusters in which you plan to deploy K8ssandra Operator.
+
+#### Generate a kubeconfig file
+Many k8s distributions come with tools for generating a kubeconfig entry. 
+
+**kind**
+
+Assuming you already have a kind cluster, you can easily export the kubeconfig entry with the following command:
+
+```
+kind get kubeconfig --name <kind-cluster-name> > kubeconfig
+```
+**GKE**
+
+The `gcloud container clusters get-credentials` command will generate a kubeconfig entry. Suppose we have a cluster in the us-east1 region, and its name is k8ssandra. 
+
+First, point to a kubeconfig file that is not the default:
+
+```
+export KUBECONFIG=/tmp/kubeconfig
+```
+Then generate the entry:
+
+```
+gcloud container clusters get-credentials --region us-east1 k8ssandra
+```
+
+**EKS**
+
+TODO
+
+**AKS**
+
+TODO
+
+#### Create the kubeconfig Secret 
+
+```
+kubectl create secret generic <secret-name> --from-file=/path/to/kubeconfig
+```
+**Note:** The property in the secret must be named `kubeconfig`, so if you are creating the secret with the `--from-file` option, then the file must also be named `kubeconfig`.
+
+#### Create the ClientConfig
+Here is an example ClientConfig manifest:
+
+```
+apiVersion: k8ssandra.io/v1alpha1
+kind: ClientConfig
+metadata:
+  name: cluster-1
+spec:
+  contextName: cluster-1
+  kubeConfigSecret:
+    name: k8s-secret  
+```
+
+`contextName` specifies the name of the Kubernetes context for which client connection should be made.
```

Comment:
>Since the general recommendation is to create separate kubeconfig files for each context/cluster that we want to manage

#55 moves us towards that approach, but I am not yet convinced that a separate kubeconfig file per cluster is easier. It is worth noting that multiple ClientConfigs can reference the same secret.

>couldn't we simply create a connection to all contexts specified in the file

We definitely could and that would probably make things easier. 

Thinking out loud... we could change the ClientConfig to make the `KubeConfigSecret` property a slice so that it can specify multiple contexts. If it is empty, we create clients for all contexts; otherwise, we create clients only for those specified. Thoughts?

Reply:
> we could change the ClientConfig to make the KubeConfigSecret property a slice so that it can specify multiple contexts. If it is empty, we create clients for all contexts; otherwise, we create clients only for those specified. 

+1, but maybe we should wait for #82 to be fixed before changing ClientConfig?